temporalio/worker/_activity.py

```diff
@@ -490,6 +490,18 @@ async def _run_activity(
                         temporalio.exceptions.CancelledError("Cancelled"),
                         completion.result.cancelled.failure,
                     )
+                elif (
+                    isinstance(
+                        err,
+                        temporalio.exceptions.ApplicationError,
+                    )
+                    and err.category
+                    == temporalio.exceptions.ApplicationErrorCategory.BENIGN
+                ):
+                    # Do not log for ApplicationError with BENIGN category.
+                    await self._data_converter.encode_failure(
+                        err, completion.result.failed.failure
+                    )
```

Comment:
Can we move just the skipping (or level changing) of the log to the `else` below but keep the rest of the logic down there? It may just be the encode failure (and a what-will-be-ignored broken executor) thing right now, but since this only affects logging it will be more logical if this conditional only governs the logging portion and doesn't get its own error encoding arm (even though some of the ones above can seem that way, they are more).

Reply:
Done.